frontend_app_file/src/component/Kanban.jsx

```diff
@@ -0,0 +1,386 @@
+import React from 'react'
+import { translate } from 'react-i18next'
+import { v4 as uuidv4 } from 'uuid'
+import Board from '@lourenci/react-kanban'
+import '@lourenci/react-kanban/dist/styles.css'
+import {
+  IconButton,
+  PromptMessage,
+  handleFetchResult,
+  putFileContent,
+  removeLocalStorageItem,
+  getLocalStorageItem,
+  setLocalStorageItem
+} from 'tracim_frontend_lib'
+
+require('./Kanban.styl')
+
+function moveElem (array, fromPosition, toPosition, elem) {
+  const a = array.slice()
+  a.splice(fromPosition, 1)
+  a.splice(toPosition, 0, elem)
+  return a
+}
+
+export default translate()(class Kanban extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      loading: false,
+      mustSave: false,
+      board: {
+        columns: [],
+        ...JSON.parse(props.content.raw_content || '{}')
+      },
+      isDrafAvailable: !!getLocalStorageItem(
+        'rawContent',
+        props.content,
+        'kanban'
+      )
+    }
+  }
+
+  componentDidUpdate (prevProps) {
+    if (this.props.content.current_revision_id !== prevProps.content.current_revision_id) {
+      // TODO handle the "Refresh?" button
+      this.updateBoard({
+        ...this.state.board,
+        ...JSON.parse(this.props.content.raw_content || '{}')
+      }, true)
+    }
+  }
+
+  handleRestoreDraft = () => {
+    this.updateBoard(
+      JSON.parse(getLocalStorageItem('rawContent', this.props.content, 'kanban')),
+      true
+    )
+    this.setState({ isDrafAvailable: false, mustSave: true })
+  }
+
+  handleDiscardChanges = () => {
+    if (confirm(this.props.t('Are you sure you want to discard your changes?'))) {
+      this.updateBoard(JSON.parse(this.props.content.raw_content || '{}'), true)
+      removeLocalStorageItem('rawContent', this.props.content, 'kanban')
+      this.setState({ mustSave: false, isDraftAvailable: false })
+    }
+  }
+
+  handleEditCardTitle = (card) => {
+    const title = prompt(this.props.t('Please enter the new title of this card'), card.title || '')
+    if (title) {
+      this.updateCard(card, { ...card, title })
+    }
+  }
+
+  handleEditCardContent = (card) => {
+    const description = prompt(this.props.t('Please enter the new content of this card'), card.description || '')
+    if (description) {
+      this.updateCard(card, { ...card, description })
+    }
+  }
+
+  updateCard (oldCard, updatedCard) {
+    this.updateColumns(
+      this.state.board.columns.map(
+        col => ({
+          ...col,
+          cards: col.cards.map(
+            card => (
+              oldCard === card
+                ? updatedCard
+                : card
+            )
+          )
+        })
+      )
+    )
+  }
+
+  removeCard = (card) => {
+    this.updateColumns(
+      this.state.board.columns.map(
+        col => ({
+          ...col,
+          cards: col.cards.filter(c => c !== card)
+        })
+      )
+    )
+  }
+
+  renderCard = (card, arg2, arg3) => {
```

Comment:
arg2 and arg3 seems unused

Reply:
```suggestion
  renderCard = (card) => {
```

Reply:
Yes, this is a leftover for when I was guessing the API